packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx

```diff
@@ -158,13 +145,28 @@ export const ConfigurationPage = (props: ConfigurationPageProps): JSX.Element =>
     microphone: PermissionState;
   } = {
     // fall back to using cameraPermissionGranted and microphonePermissionGranted if permission API is not supported
-    camera: videoState ?? (cameraPermissionGranted ? 'granted' : 'denied'),
-    microphone: audioState ?? (microphonePermissionGranted ? 'granted' : 'denied')
+    camera: videoState && videoState !== 'unsupported' ? videoState : cameraPermissionGranted ? 'granted' : 'denied',
+    microphone:
+      audioState && audioState !== 'unsupported' ? audioState : microphonePermissionGranted ? 'granted' : 'denied'
   };
   /* @conditional-compile-remove(call-readiness) */
   const networkErrors = errorBarProps.activeErrorMessages.filter((message) => message.type === 'callNetworkQualityLow');
+
   /* @conditional-compile-remove(call-readiness) */
-  const [isModalShowing, setIsModalShowing] = useState(true);
+  const [isPermissionsModalDismissed, setIsPermissionsModalDismissed] = useState(true);
+
+  // When permission API is not available, we want to show screen saying checking for access (disappears on its own)
+  // then based on permission setting, we show permission denied or nothing
+  /* @conditional-compile-remove(call-readiness) */
+  const [checkPermissionModalShowing, setCheckPermissionModalShowing] = useState(true);
```

Comment:
`setCheckPermissionModalShowing` and `permissionDeniedModalShowing` are always inverse of each other (and hence directly correlated). To reduce redundancy can we combine these variables into:

```ts
const [minimumFallbackTimerElapsed, setMinimumFallbackTimerElapsed] = useState(false);
```

then for below it becomes:
```tsx

const forceShowingCheckPermissions = !minimumFallbackTimerElapsed;

<CallReadinessModalFallBack
  // rename this to forceCheckPermissionsModalShowing as this can show even
  // when the prop is false (when the permissions aren't known yet due to system prompt)
  checkPermissionModalShowing={forceShowingCheckPermissions}
/>
```

and then remove the `permissionDeniedModalShowing` param as this will be determined based on the other props passed in